File: Linked-list/Single_linked_list.py
class node:
    
    def __init__(self, value):
        self.value = value
        self.nextnode = None


class linked_list:

    def __init__(self, head = None):
        self.head = head

    def print_linked_list(self):
        node_iterator = self.head
        while node_iterator != None:
            print(node_iterator.value)
            node_iterator = node_iterator.nextnode

    def insert(self, value):
        new_node = node(value)
        if self.head == None:
            self.head = new_node
        else:
            node_iterator = self.head
            while node_iterator.nextnode != None:
                node_iterator = node_iterator.nextnode
            node_iterator.nextnode = new_node

    def delete_node(self, index):
        i = 0
        node_iterator = self.head
        while node_iterator.nextnode != None and i != index - 1:
            node_iterator = node_iterator.nextnode
            i += 1
        node_iterator.nextnode = node_iterator.nextnode.nextnode

demo_linked_list = linked_list()
for i in range(10):
    demo_linked_list.insert(i)
demo_linked_list.print_linked_list()

File: Linked-list/Single_linked_list.c
#include <stdio.h>
#include <stdlib.h>

struct Node
{
    int data;
    struct Node* next;
};

int print_linked_list(struct Node* head)
{
    struct Node* curr = head;
    printf("%d %p %p\n\n", head->data, (void*)&head, (void*)head->next);
    while (curr != NULL) {
        printf("Val: %d   Addr: %p   Next: %p\n", curr->data, (void*)curr, (void*)curr->next);
        curr = curr->next;
    }
    printf("Over");
    return 0;
    
}

int main()
{
    struct Node* head = NULL;
    int arr[10] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    int len = sizeof(arr) / sizeof(arr[0]);
    for (int i=0; i<len; i++) {
        struct Node* newnode = (struct Node*)malloc(sizeof(struct Node));
        newnode->data = arr[i];
        newnode->next = NULL;
        if (head == NULL) {
            head = newnode;
        }
        else {
            struct Node* curr = head;
            while (curr->next != NULL) {
                curr = curr->next;
            }
            curr->next = newnode;
        }
        printf("%d inserted\n", arr[i]);
    }
    printf("--- --- ---\n");
    print_linked_list(head);
    return 0;
}